server/mailet/integration-testing/src/test/java/org/apache/james/mailets/ForwardLoopIntegrationTest.java

```diff
@@ -158,15 +159,15 @@ void filterForwardShouldNotCreateLoopError() throws Exception {
             .authenticate(SENDER.asString(), PASSWORD)
             .sendMessage(SENDER.asString(), ALICE.asString());
 
-        Awaitility.waitAtMost(1L, TimeUnit.HOURS).until(() -> mailRepositoryProbe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 2L);
+        Awaitility.await().until(() -> jamesServer.getProbe(SpoolerProbe.class).processingFinished());
+        Awaitility.await().until(() -> mailRepositoryProbe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 1L);
 
         SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
             List<Mail> mails = mailRepositoryProbe.listMails(CUSTOM_REPOSITORY)
                 .filter(Throwing.predicate(mail -> mail.getRecipients().contains(ALICE.asMailAddress())))
                 .collect(ImmutableList.toImmutableList());
 
             softly.assertThat(mails.get(0).getRecipients()).containsOnly(ALICE.asMailAddress());
-            softly.assertThat(mails.get(1).getRecipients()).containsOnly(ALICE.asMailAddress());
```

Comment:
Hmmm as discussed, not ok IMO...

Meaning in the case here that Alice does not keep a local copy, the mail would just not be stored anywhere and vanish. This is unacceptable for a mail server in any way! (I'm sure @chibenwa agrees on that^^)

I think we need to think a bit deeper here. Of course we want the mail to be stored somewhere. What do we want though?
- mail stop at the end of the chian before looping (here stored on cedric mailbox)
- mail is stored back at the original recipient when loop is detected (alice, which was the behavior before I asked the change)
- if stored at the original recipient and that recipient had localCopy enabled, how do we deal with that? I don't think we should store it twice right? (same in the case of cedric if he had localCopy, we don't want to store it twice either)

Reply:
I don't think the mail is lost but it is possible that in case of loop the mail do not end up in the correct place...

Reply:
> I don't think the mail is lost but it is possible that in case of loop the mail do not end up in the correct place...
> 
> Let's write more readable test cases then see if the behaviour is the one that we want. Today it is not that clear to me to be honnest...

If we setup in the way that A forward mail to B, B forward to C and C forward to A but no one set keepACopy to be true, the mail would be lost.

Reply:
> If we setup in the way that A forward mail to B, B forward to C and C forward to A but no one set keepACopy to be true, the mail would be lost.

How about C is the end of the loop and C is the one who keeps the final mail?

Even if C does not want to `keepACopy`, standing as a user, I say when forwarding me this mail, please do not keep the mail for me. In the case that C does not forward the mail to A, it means there is no forwarding done, therefore no need to obey `keepACopy` for C which is reasonable IMO.

Maybe `Bounce` a mail to announce C that his mail can not be forward anymore as the end of the loop as well?